### PR TITLE
cleanup: change to common's {Connection,Tracing}Options

### PR DESCRIPTION
**Pull request by devjgm**
Fixes: #1260
Fixes: #1251

This PR removes most tests that are now unnecessary. I kept some of the tests in `connection_options_test.cc`, because these are the tests that verify properties that are configured by Spanner's custom Traits class.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/1279)

### CODE REVIEW

File: google/cloud/spanner/tracing_options.h [hunk 1]
```diff
@@ -16,6 +16,7 @@
 #define GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_TRACING_OPTIONS_H
 
 #include "google/cloud/spanner/version.h"
+#include "google/cloud/tracing_options.h"
 #include <cstdint>
```
Comment (devbww) [inline google/cloud/spanner/tracing_options.h]: Unused.

File: google/cloud/spanner/tracing_options.h [hunk 2]
```diff
@@ -16,6 +16,7 @@
...
 #include <string>
```
Comment (devbww) [inline google/cloud/spanner/tracing_options.h]: Unused.

File: google/cloud/spanner/tracing_options.h (hunk 1 above)
Comment (devjgm) [reply google/cloud/spanner/tracing_options.h]: Removed

File: google/cloud/spanner/tracing_options.h (hunk 2 above)
Comment (devjgm) [reply google/cloud/spanner/tracing_options.h]: removed

File: google/cloud/spanner/CMakeLists.txt [hunk 3]
```diff
@@ -300,7 +299,6 @@ function (spanner_client_define_tests)
 
     set(spanner_client_unit_tests
         # cmake-format: sortable
-        connection_options_test.cc
```
Comment (tmatsuo) [inline google/cloud/spanner/CMakeLists.txt]: Shouldn't we keep this test?

Comment (devjgm) [reply google/cloud/spanner/CMakeLists.txt]: Yes, you're right. Good catch. I accidentally deleted that one. PR coming up.

**Pull request merged**